pkg/utils/release/controller.go

```diff
@@ -394,3 +400,119 @@ func (s *SuiteController) GetSbomPyxisByImageID(pyxisStageURL, imageID string,
 	}
 	return body, nil
 }
+
+// DeleteAllSnapshotEnvironmentBindings removes all CRs SnapshotEnvronmentBindings in cluster.
+func (s *SuiteController) DeleteAllSnapshotEnvironmentBindings() error {
+	snapshotEnvironmentBindingsList := &appstudioApi.SnapshotEnvironmentBindingList{}
+
+	err := s.KubeRest().List(context.Background(), snapshotEnvironmentBindingsList, &client.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to delete all SnapshotEnvironmentBindings : %v", err)
+	}
+
+	for _, snapshotEnvironmentBinding := range snapshotEnvironmentBindingsList.Items {
+		err := wait.PollImmediate(time.Second, 30*time.Second, func() (done bool, err error) {
+			snapshotEnvironmentBindingCR := appstudioApi.SnapshotEnvironmentBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      snapshotEnvironmentBinding.Name,
+					Namespace: snapshotEnvironmentBinding.Namespace,
+				},
+			}
+			if err := s.KubeRest().Get(context.TODO(), client.ObjectKeyFromObject(&snapshotEnvironmentBindingCR), &snapshotEnvironmentBindingCR); err != nil {
+				if errors.IsNotFound(err) {
+					// napshotEnvironmentBinding CR is already removed
+					return true, nil
+				}
+				g.GinkgoWriter.Printf("unable to retrieve napshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+
+			}
+
+			// Remove the finalizer, so that it can be deleted.
+			snapshotEnvironmentBindingCR.Finalizers = []string{}
+			if err := s.KubeRest().Update(context.TODO(), &snapshotEnvironmentBindingCR); err != nil {
+				g.GinkgoWriter.Printf("unable to remove finalizers from snapshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+			}
+
+			if err := s.KubeRest().Delete(context.TODO(), &snapshotEnvironmentBindingCR); err != nil {
+				g.GinkgoWriter.Printf("unable to delete napshotEnvironmentBinding '%s' in '%s': %v\n", snapshotEnvironmentBindingCR.Name, snapshotEnvironmentBindingCR.Namespace, err)
+				return false, nil
+			}
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("deletion of napshotEnvironmentBinding '%s' in '%s' timed out", snapshotEnvironmentBinding.Name, snapshotEnvironmentBinding.Namespace)
+		}
+
+	}
+
+	return nil
+}
+
+func (s *SuiteController) WaitForReleasePipelineToBeFinished(namespace string, maxRetries int) error {
```

Comment:
I think we need to be more specific here. In case we'd have a test where we have more than 1 release pipelinerun, this method wouldn't help us in selecting which one we want to check.

I'd suggest using parameters that this method uses
https://github.com/redhat-appstudio/e2e-tests/blob/9a422b2094d75e8a2abf7ff991143930d6625cef/pkg/utils/release/controller.go#L120

Reply:
IMO rather than creating 2 methods to do the similar thing, creating one method that'd be robust to fit both cases (1 or 2 release pipelines per namespace) makes more sense.

take for example similar method in integration controller:
https://github.com/redhat-appstudio/e2e-tests/blob/9a422b2094d75e8a2abf7ff991143930d6625cef/pkg/utils/integration/controller.go#L372
and in has controller:
https://github.com/redhat-appstudio/e2e-tests/blob/9a422b2094d75e8a2abf7ff991143930d6625cef/pkg/utils/has/controller.go#L496

which target a specific pipelinerun based on associated resources (component/snapshot etc.)